impeller/entity/geometry.cc

```diff
@@ -156,4 +172,130 @@ std::optional<Rect> VerticesGeometry::GetCoverage(Matrix transform) {
+    std::shared_ptr<Tessellator> tessellator,
+    Color paint_color,
+    BlendMode blend_mode) {
+  // TODO(jonahwilliams): support per-color vertex in path geometry.
+  return {};
+}
+
+GeometryResult PathGeometry::GetPositionUVBuffer(
+    std::shared_ptr<Allocator> device_allocator,
+    HostBuffer& host_buffer,
+    std::shared_ptr<Tessellator> tessellator,
+    ISize render_target_size) {
+  // TODO(jonahwilliams): support texture coordinates in path geometry.
+  return {};
+}
+
+GeometryVertexType PathGeometry::GetVertexType() {
+  return GeometryVertexType::kPosition;
+}
+
+std::optional<Rect> PathGeometry::GetCoverage(Matrix transform) {
+  return path_.GetTransformedBoundingBox(transform);
+}
+
+/////// Cover Geometry ///////
+
+CoverGeometry::CoverGeometry() {}
+
+CoverGeometry::~CoverGeometry() {}
+
+GeometryResult CoverGeometry::GetPositionBuffer(
+    std::shared_ptr<Allocator> device_allocator,
+    HostBuffer& host_buffer,
+    std::shared_ptr<Tessellator> tessellator,
+    ISize render_target_size) {
+  // TODO(jonahwilliams): remove tessellation and just write rect into buffers.
+  auto path = PathBuilder{}.AddRect(Size(render_target_size)).TakePath();
+  VertexBuffer vertex_buffer;
+  auto tesselation_result = tessellator->TessellateBuilder(
+      path.GetFillType(), path.CreatePolyline(),
+      [&vertex_buffer, &host_buffer](
+          const float* vertices, size_t vertices_count, const uint16_t* indices,
+          size_t indices_count) {
+        vertex_buffer.vertex_buffer = host_buffer.Emplace(
+            vertices, vertices_count * sizeof(float), alignof(float));
+        vertex_buffer.index_buffer = host_buffer.Emplace(
+            indices, indices_count * sizeof(uint16_t), alignof(uint16_t));
+        vertex_buffer.index_count = indices_count;
+        vertex_buffer.index_type = IndexType::k16bit;
+        return true;
+      });
+  if (tesselation_result != Tessellator::Result::kSuccess) {
+    return {};
+  }
+  return GeometryResult{
+      .type = PrimitiveType::kTriangle,
+      .vertex_buffer = vertex_buffer,
+      .prevent_overdraw = false,
+  };
+}
+
+GeometryResult CoverGeometry::GetPositionColorBuffer(
+    std::shared_ptr<Allocator> device_allocator,
+    HostBuffer& host_buffer,
+    std::shared_ptr<Tessellator> tessellator,
+    Color paint_color,
+    BlendMode blend_mode) {
+  // TODO(jonahwilliams): support per-color vertex in cover geometry.
+  return {};
+}
+
+GeometryResult CoverGeometry::GetPositionUVBuffer(
+    std::shared_ptr<Allocator> device_allocator,
+    HostBuffer& host_buffer,
+    std::shared_ptr<Tessellator> tessellator,
+    ISize render_target_size) {
+  // TODO(jonahwilliams): support texture coordinates in cover geometry.
+  return {};
+}
+
+GeometryVertexType CoverGeometry::GetVertexType() {
+  return GeometryVertexType::kPosition;
+}
+
+std::optional<Rect> CoverGeometry::GetCoverage(Matrix transform) {
+  return Rect::MakeLTRB(-INFINITY, -INFINITY, INFINITY, INFINITY);
```

Comment:
crowdsourcing some alternative ideas here 😄

Reply:
Handling infinity rects has been a long time coming. I believe infinity coverage shouldn't cause a problem for EntityPass in its current given the frustum/stencil clip cropping. We've already got a few Aiks tests using DrawPaint -- including ColorWheel. If none of these tests crash when flipping through them then we're probably good to go.

Reply:
Ah, I just noticed this doesn't replace `DrawPaint` yet. I'd say this is good. If we find problems down the road when landing it they should be easy fixes. It also occurs to me that we still need move comprehensive filter cropping support.
Although eventually we should eventually just ignore most image filters and execute color filters on the CPU for the `DrawPaint` case.

Reply:
👍